controllers/releaseplanadmission/enqueue_status.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package releaseplanadmission
+
+import (
+	"context"
+	"strings"
+
+	"github.com/redhat-appstudio/release-service/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	crtHandler "sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var log = logf.Log.WithName("event_handler")
+
+// EnqueueForReleasePlanStatus enqueues Request containing the Name and Namespace specified in the
+// Status.ReleasePlanAdmission of the ReleasePlan that is the source of the Event. The source of the event
+// triggers reconciliation of the parent resource.
+type EnqueueForReleasePlanStatus struct {
```

Comment:
We should have a single `EnqueueForPairingUpdate`. That's why we pass the type here, to know what we are calling. The code becomes a bit more complex, but more centralized.

Reply:
I would probably move these files to a different package (eg. `controllers/utils/handlers`). The predicates one I would move it to `controllers/utils/predicates`. Ideally I would move it inside the specific type, but they are shared, so... this looks fine.

Reply:
moved. much better